lib/shared-method.js

```diff
@@ -440,16 +444,17 @@ function convertValueToTargetType(argName, value, targetType) {
  * @returns {String} The type name
  */
 
-SharedMethod.getType = function(val) {
+SharedMethod.getType = function(val, targetType) {
   var type = typeof val;
 
   switch (type) {
     case 'undefined':
     case 'boolean':
-    case 'number':
     case 'function':
     case 'string':
       return type;
+    case 'number':
+      return Number.isInteger(val) ? (targetType || 'integer') : 'number';
```

Comment:
that's the hacky way to work with numbers which are of pattern: `x.0`, for example: `2.0`